flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1828,6 +1828,33 @@ class JobManager(
     jobManagerMetricGroup.gauge[Long, Gauge[Long]]("numRunningJobs", new Gauge[Long] {
       override def getValue: Long = JobManager.this.currentJobs.size
     })
+    jobManagerMetricGroup.gauge[Long, Gauge[Long]]("numFailedJobs", new Gauge[Long] {
+      override def getValue: Long = {
+         var failedJobs = 0
+         val ourJobs = createJobStatusOverview()
+         val future = (archive ? RequestJobsOverview.getInstance())(timeout)
+         val archivedJobs : JobsOverview = Await.result(future, timeout).asInstanceOf[JobsOverview]
+         failedJobs += ourJobs.getNumJobsFailed() + archivedJobs.getNumJobsFailed()
+         failedJobs
+    }})
+    jobManagerMetricGroup.gauge[Long, Gauge[Long]]("numCancelledJobs", new Gauge[Long] {
+      override def getValue: Long = {
+         var cancelledJobs = 0
+         val ourJobs = createJobStatusOverview()
+         val future = (archive ? RequestJobsOverview.getInstance())(timeout)
+         val archivedJobs : JobsOverview = Await.result(future, timeout).asInstanceOf[JobsOverview]
+         cancelledJobs += ourJobs.getNumJobsCancelled() + archivedJobs.getNumJobsCancelled()
+         cancelledJobs
+    }})
+    jobManagerMetricGroup.gauge[Long, Gauge[Long]]("numFinishedJobs", new Gauge[Long] {
+      override def getValue: Long = {
+         var finishedJobs = 0
+         val ourJobs = createJobStatusOverview()
+         val future = (archive ? RequestJobsOverview.getInstance())(timeout)
+         val archivedJobs : JobsOverview = Await.result(future, timeout).asInstanceOf[JobsOverview]
+         finishedJobs += ourJobs.getNumJobsFinished() + archivedJobs.getNumJobsFinished()
+         finishedJobs
+    }})
```

Comment:
@zentol , @rmetzger not sure if you meant something like to following perhaps to work around the RPC calls (I guess this would eliminate them completely however are the changes a bit more "invasive" I would think):
So what we could do is where we receive the Job status change
```
case JobStatusChanged(jobID, newJobStatus, timeStamp, error) =>
```
Send a RemoveJob message with the newJobStatus (could be "cancelled" as far as I can see)
```
self ! decorateMessage(RemoveJob(jobID, removeJobFromStateBackend = true, newJobStatus))
```
Then for the RemoveJob message propagate this to the method that actually removes the job:
```
removeJob(graph.getJobID, clearPersistedJob, newJobStatus)
```
And finally besides taking the job out of the map representing the currentJobs maybe increment a counter for canceled archived jobs or have a different map that stores archived canceled jobs (and others for failed and finished) so that when it is time for sending out the monitoring metrics we would use this map/counter rather than doing any RPC calls at all.
What do you guys think ?